src/run/run.c

```diff
@@ -1721,6 +1741,10 @@ static int run(int argc, char* argv[]) {
         if (r < 0)
                 return r;
 
+        if (retval < 0)
+                /* propogate the fatal signal to the caller */
+                arg_kill_signal = -retval;
+
         return retval;
```

Comment:
hmm, i figure we should do "return retval < 0 ? EXIT_EXCEPTION : retval;" or so here, otherwise we'll sometimes return the signal number literally as exit code, which I think isn't the right approach...